Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift

```diff
@@ -24,37 +30,145 @@ class MessageListVC<ExtraData: ExtraDataTypes>: _ViewController, UICollectionVie
+    
+    private var messageComposerBottomConstraint: NSLayoutConstraint?
+    
+    private var timer: Timer?
+    
+    private lazy var keyboardObserver = ChatMessageListKeyboardObserver(
+        containerView: view,
+        scrollView: collectionView,
+        composerBottomConstraint: messageComposerBottomConstraint
+    )
+
+    public lazy var userSuggestionSearchController: _ChatUserSearchController<ExtraData> = {
+        channelController.client.userSearchController()
+    }()
+    
+    // Load from UIConfig
+    public lazy var titleView = ChatMessageListTitleView<ExtraData>()
+    
+    public lazy var impactFeedbackGenerator = UIImpactFeedbackGenerator(style: .medium)
+
+//    public lazy var router = uiConfig.navigation.messageListRouter.init(rootViewController: self)
+    public lazy var router = MessageListRouter(rootViewController: self)
+    
+    override func setUp() {
+        super.setUp()
+        
+        messageComposerViewController.delegate = .wrap(self)
+        messageComposerViewController.controller = channelController
+        messageComposerViewController.userSuggestionSearchController = userSuggestionSearchController
+
+        userSuggestionSearchController.search(term: nil)
+        
+        channelController.setDelegate(self)
+        channelController.synchronize()
+        
+        if channelController.isChannelDirect {
+            timer = Timer.scheduledTimer(withTimeInterval: 60, repeats: true) { [weak self] _ in
+                self?.updateNavigationTitle()
+            }
+        }
+    }
+    
+    override func setUpLayout() {
+        super.setUpLayout()
+        
+        view.addSubview(collectionView)
+        collectionView.pin(anchors: [.top, .leading, .trailing], to: view.safeAreaLayoutGuide)
+        
+        messageComposerViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        addChildViewController(messageComposerViewController, targetView: view)
+
+        messageComposerViewController.view.topAnchor.pin(equalTo: collectionView.bottomAnchor).isActive = true
+        messageComposerViewController.view.leadingAnchor.pin(equalTo: view.safeAreaLayoutGuide.leadingAnchor).isActive = true
+        messageComposerViewController.view.trailingAnchor.pin(equalTo: view.safeAreaLayoutGuide.trailingAnchor).isActive = true
+        messageComposerBottomConstraint = messageComposerViewController.view.bottomAnchor.pin(equalTo: view.bottomAnchor)
+        messageComposerBottomConstraint?.isActive = true
+    }
+    
+    override func defaultAppearance() {
+        super.defaultAppearance()
+        
+        view.backgroundColor = .white
         
-        view.embed(collectionView)
         collectionView.backgroundColor = .white
         
-        collectionView.register(MessageCell<ExtraData>.self, forCellWithReuseIdentifier: MessageCell<ExtraData>.reuseId)
+        navigationItem.titleView = titleView
+    }
+    
+    private func updateNavigationTitle() {
```

Comment:
this needs to be called when something in the channel changes, too. You can just listen to `channel controller(didUpdateChannel:)` callbacks and call this.

Reply:
✅